DuckDuckGo/VPNShortcutIntents.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  VPNShortcutIntents.swift
+//  DuckDuckGo
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import AppIntents
+import NetworkExtension
+import NetworkProtection
+import WidgetKit
+import Core
+import VPNWidgetSupport
+
+// MARK: - Enable & Disable
+
+/// App intent to disable the VPN
+///
+/// This is used in App Shortcuts, for things like Shortcuts.app, Spotlight and Siri.
+/// This is very similar to ``WidgetDisableVPNIntent``, but this runs in-app, allows continuation in the app if needed,
+/// and provides a result dialog.
+///
+@available(iOS 17.0, *)
+struct DisableVPNIntent: AppIntent {
+
+    private enum DisableAttemptFailure: CustomNSError {
+        case cancelled
+    }
+
+    static let title: LocalizedStringResource = "Disable DuckDuckGo VPN"
+    static let description: LocalizedStringResource = "Disables the DuckDuckGo VPN"
```

Comment:
Should these also be handled in a localized way?

Reply:
Localized.

Reply:
I had to roll these back, because it's not easy to map our existing localization mechanism to `LocalizedStringResource`.  The compiler complains about the passed string not being a literal.

That said these are localized string resources so we'll need to figure out how to map these to our regular mechanism.

FWIW though I think that's exposed is the shortcut name, and not the intent name.

Reply:
Added a second bullet [here](https://app.asana.com/0/1206580121312550/1209026082513680/f).